forge-core/src/main/java/forge/util/collect/FCollection.java

```diff
@@ -43,12 +42,38 @@ public static <T> FCollection<T> getEmpty() {
     /**
      * The {@link Set} representation of this collection.
      */
-    private final Set<T> set = Sets.newHashSet();
+    private Set<T> SET;
+    private Set<T> set() {
+        Set<T> result = SET;
+        if (result == null) {
+            synchronized (this) {
+                result = SET;
+                if (result == null) {
+                    result = Sets.newConcurrentHashSet();
+                    SET = result;
+                }
+            }
+        }
+        return SET;
+    }
 
     /**
      * The {@link List} representation of this collection.
      */
-    private final LinkedList<T> list = Lists.newLinkedList();
+    private List<T> LIST;
+    private List<T> list() {
+        List<T> result = LIST;
+        if (result == null) {
+            synchronized (this) {
+                result = LIST;
+                if (result == null) {
+                    result = Lists.newCopyOnWriteArrayList();
```

Comment:
by switching the default iterator you will produce crashes imo

because some code uses add() or remove()
and LinkedList did support it...

but read JavaDoc of `CopyOnWriteArrayList`

```
 * The iterator will not reflect additions, removals, or changes to
 * the list since the iterator was created.  Element-changing
 * operations on iterators themselves ({@code remove}, {@code set}, and
 * {@code add}) are not supported. These methods throw
 * {@code UnsupportedOperationException}.
```

Reply:
In most of our use cases, we need to make a copy via threadSafeIterable anyway, 
otherwise, we get `ConcurrentModificationException`

And the Test seems to pass so far?

Reply:
I trust my brain/IDE more here ;)

Here's one example from PlayAi:
![image](https://github.com/user-attachments/assets/5ed4819c-b806-456e-8946-69c67fb5400a)

this will call remove() on a CardCollection, which is a FCollection
I don't see a way this doesn't crash now

Reply:
imo a better way would be to get the ones into a `toRemove` collection, and then remove them after the loop?

Reply:
> I trust my brain/IDE more here ;)
> 
> Here's one example from PlayAi: ![image](https://private-user-images.githubusercontent.com/8506892/386260457-5ed4819c-b806-456e-8946-69c67fb5400a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE2MTcyMDUsIm5iZiI6MTczMTYxNjkwNSwicGF0aCI6Ii84NTA2ODkyLzM4NjI2MDQ1Ny01ZWQ0ODE5Yy1iODA2LTQ1NmUtODk0Ni02OWM2N2ZiNTQwMGEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTRUMjA0MTQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmQzNzcxZDQ0YTY3NGY1ZWFmYjUyYzI4MDBkN2Q1OTdhMmQ0MTFlYTlhYmM4NTVjZmJmMTI5OTg2OTQ3OWNhMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ._bCvRzpWkdKaPXedtVQTciX5zqs8QJaGi2oYvZveTmI)
> 
> this will call remove() on a CardCollection, which is a FCollection I don't see a way this doesn't crash now

Can you write a simple test so it will crash if the iterator add or remove is not supported?

Reply:
```
    @Test
    void testBadIteratorLogic() {
        CardCollection cc = new CardCollection(new Card(1, null));
        Iterator<Card> it = cc.iterator();
        it.next();
        it.remove();
    }
```

Reply:
ah yes, i see it so does card collection use this? the docs says use list.remove() but I think i need to update those unless I can change to different concurrentlist which isn't available on default libs

Reply:
I suppose the direct calls have been avoided, I probably don't know enough about concurrency to consider any other implications from this 🤷‍♂️